website/addons/wiki/views.py

```diff
@@ -439,3 +443,115 @@ def project_wiki_validate_name(wname, **kwargs):
+        component_wiki_pages = {
+            'title': 'Component Wiki Pages',
+            'kind': 'folder',
+            'type': 'heading',
+            'children': format_component_wiki_pages(node, auth)
+        }
+        pages.append(component_wiki_pages)
+
+    return pages
+
+
+def format_home_wiki_page(node):
+    home_wiki = node.get_wiki_page('home')
+    home_wiki_page = {
+        'page': {
+            'url': node.web_url_for('project_wiki_home'),
+            'name': 'Home',
+            'id': 'None',
+            'wiki_content': ''
+        }
+    }
+    if home_wiki:
+        home_wiki_page = {
+            'page': {
+                'url': node.web_url_for('project_wiki_view', wname='home', _guid=True),
+                'name': 'Home',
+                'id': home_wiki._primary_key,
+                'wiki_content': wiki_page_content(home_wiki.page_name, node=node).get('wiki_content')
+            }
+        }
+    return home_wiki_page
+
+
+def format_project_wiki_pages(node, auth):
+    pages = []
+    can_edit = node.has_permission(auth.user, 'write') and not node.is_registration
+    project_wiki_pages = _get_wiki_pages_current(node)
+    home_wiki_page = format_home_wiki_page(node)
+    pages.append(home_wiki_page)
+    for wiki_page in project_wiki_pages:
+        if wiki_page['name'] != 'home':
+            wiki_content = wiki_page['wiki_content'].get('wiki_content')
+            page = {
+                'page': {
+                    'url': wiki_page['url'],
+                    'name': wiki_page['name'],
+                    'id': wiki_page['wiki_id'],
+                    'wiki_content': wiki_content
+                }
+            }
+            if can_edit or wiki_content is not '':
+                pages.append(page)
+    return pages
+
+
+def format_component_wiki_pages(node, auth):
+    pages = []
+    can_edit = node.has_permission(auth.user, 'write') and not node.is_registration
+    component_wiki_pages = _serialize_wiki_toc(node, auth)
+    for wiki_page in component_wiki_pages:
+        children = []
+        wiki_content = wiki_page['wiki_content'].get('wiki_content')
+        component_home_wiki = {
+            'page': {
+                'url': wiki_page['url'],
+                'name': 'Home',
+                'id': wiki_page['id'],
+                'wiki_content': wiki_content
+            }
+        }
+        if can_edit or wiki_content is not '':
+            children.append(component_home_wiki)
+
+        for component_page in wiki_page['pages_current']:
+            if component_page['name'] != 'home':
+                nested_wiki_content = component_page['wiki_content'].get('wiki_content')
+                child = {
+                    'page': {
+                        'url': component_page['url'],
+                        'name': component_page['name'],
+                        'id': component_page['wiki_id'],
+                        'wiki_content': nested_wiki_content
```

Comment:
Is the `wiki_content` actually used on the client side? These might contain a lot of text, which will unnecessarily bloat the response body.